src/main/java/com/blackducksoftware/ohcount4j/detect/AbstractExtnResolver.java

```diff
@@ -32,11 +32,18 @@ protected String headContent(SourceFile sourceFile) throws IOException {
     }
 
     protected CharSequence getCharSequence(SourceFile sourceFile) throws IOException {
-        return sourceFile.getCharSequence();
+        CharSequence charSequence = sourceFile.getCharSequence();
+		return charSequence != null ? charSequence : new String();
     }
 
     protected char[] getContents(SourceFile sourceFile) throws IOException {
-        return SourceFileUtils.getContents(sourceFile);
+        char[] contents = SourceFileUtils.getContents(sourceFile);
```

Comment:
This should be changed on the SourceFileUtils.getContents side so that the "getContents" method *always* returns a valid result -- either a `char []` with the contents of the file or an empty char array (`new char[0]`).

That way the `getContentsAsString` can be simplified:

```Java
protected String getContentsAsString(SourceFile sourceFile) throws IOException {
  return new String(SourceFileUtils.getContents(sourceFile);
}
```
And this way we don't have three nearly identical checks for the same type of situation.